### PR TITLE
chore(docs): re-add base config to spin.toml

### PR DESCRIPTION
**Pull request by vdice**
We're still pinned to an earlier version of spin when [publishing (via bindle)](https://github.com/fermyon/spin/blob/main/deploy/publish-spin-docs.nomad) and deploying the spin docs 'website' (comprised of redirects from spin.fermyon.dev -> developer.fermyon.com/spin), so for now, we need the `base` config.

Fixes deployment error as seen recentlly in https://github.com/fermyon/spin/actions/runs/6317502067

Otherwise, the issue to move the website deployment to cloud is https://github.com/fermyon/spin/issues/1819 (and we can then use latest spin, cloud plugin, etc via https://github.com/fermyon/actions)

### CODE REVIEW

**Comment by vdice** [comment]
(this PR updates a file in `docs` so many of the required checks aren't run, so will be stuck in pending; I can merge once an approval is in.)

**Review by itowlson — APPROVED** [review]

**Pull request merged**
